pages/email.tsx

```diff
@@ -27,6 +29,35 @@ const initialValues: EmailEsrfApiRequestBody = {
 export default function Email() {
   const { t } = useTranslation('email')
   const [modalOpen, setModalOpen] = useState(false)
```

Comment:
Here's my recommended code refactoring:

```tsx
  const router = useRouter()
  const [modalOpen, setModalOpen] = useState(false)
  const [isIdle, setIsIdle] = useState(false)

  const handleOnIdleTimerIdle = useCallback(() => {
    setIsIdle(true)
    setModalOpen(true)
  }, [])

  const { reset: resetIdleTimer } = useIdleTimer({
    onIdle: handleOnIdleTimerIdle,
    //15 minute timeout
    timeout: 15 * 60 * 1000,
  })

  const handleOnModalRedirectButtonClick = useCallback(() => {
    //If user is idle and selects option to go back, clear the cookie so they get redirected to /expectations instead
    if (isIdle) {
      deleteCookie('agreed-to-email-esrf-terms')
    }
    router.push('/landing')
  }, [isIdle, router])

  const handleOnModalResetButtonClick = useCallback(() => {
    setModalOpen(false)
    if (isIdle) {
      setIsIdle(false)
      resetIdleTimer()
    }
  }, [isIdle, resetIdleTimer])
```

The main changes are:

- Use `useRouter` hook instead of `Router`
- Change the timeout calculation from 150 * 6 * 1000` to `15 * 60 * 1000`
- Use `useCallback` hook
- Rename handler method name like `handle{EventThingNameHere}`
- Use `deleteCookie` instead of `setCookie`

Reply:
This can also be applied to `email.tsx` page.